File: learningGraph.py
n, m, k = map(int, input().split())
val = list(map(int, input().split()))
adj = [[] for _ in range(n)]
for _ in range(m):
	x, y = map(int, input().split())
	adj[x-1].append((val[y-1], y))
	adj[y-1].append((val[x-1], x))
for i in range(n):
	adj[i].sort(reverse = True)
for i in range(n):
	if len(adj[i]) < k:
		print(-1)
	else:
		print(adj[i][k-1][1])

File: learningGraph.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main(void) {
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n, m, k;
    cin >> n >> m >> k;
    vector<int> v(n);
    for (int i = 0; i < n; i++)
        cin >> v[i];
    vector<pair<int, int>> adj[n];
    for (int i = 0; i < m; i++) {
        int x, y;
        cin >> x >> y;
        adj[x-1].push_back({v[y-1], y});
        adj[y-1].push_back({v[x-1], x});
    }
    for (int i = 0; i < n; i++)
        sort(adj[i].begin(), adj[i].end());
    for (int i = 0; i < n; i++)
        (adj[i].size() < k) ? cout << -1 << endl : cout << adj[i][adj[i].size()-k].second << endl;
}
